app/services/katello/rhsm_fact_parser.rb

```diff
@@ -24,12 +24,16 @@ def support_interfaces_parsing?
     end
 
     def get_facts_for_interface(interface)
-      {
+      result = {
         'link' => true,
-        'macaddress' => get_rhsm_mac(interface),
-        'ipaddress' => get_rhsm_ip(interface),
-        'ipaddress6' => get_rhsm_ipv6(interface),
       }
+      mac = get_rhsm_mac(interface)
+      result['macaddress'] = mac if mac
+      ip = get_rhsm_ip(interface)
+      result['ipaddress'] = ip if ip
+      ipv6 = get_rhsm_ipv6(interface)
+      result['ipaddress6'] = ipv6 if ipv6
+      result
```

Comment:
Alternative is to keep the old hash and then use `{ ... }.reject { |_, value| value.nil }`.

Reply:
Looks nicer, yeah tho it allocates an extra hash. Not a big deal for this small one tho.